tests/system/providers/google/bigquery/example_bigquery_operations.py

```diff
@@ -0,0 +1,106 @@
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Example Airflow DAG for Google BigQuery service local file upload and external table creation.
+"""
+import os
+from datetime import datetime
+from pathlib import Path
+
+from airflow import models
+from airflow.providers.google.cloud.operators.bigquery import (
+    BigQueryCreateEmptyDatasetOperator,
+    BigQueryCreateExternalTableOperator,
+    BigQueryDeleteDatasetOperator,
+)
+from airflow.providers.google.cloud.operators.gcs import GCSCreateBucketOperator, GCSDeleteBucketOperator
+from airflow.providers.google.cloud.transfers.local_to_gcs import LocalFilesystemToGCSOperator
+from airflow.utils.trigger_rule import TriggerRule
+from tests.system.utils.watcher import watcher
+
+ENV_ID = os.environ.get("SYSTEM_TESTS_ENV_ID")
+DAG_ID = "bigquery_operations"
+
+DATASET_NAME = f"dataset_{DAG_ID}_{ENV_ID}"
+DATA_SAMPLE_GCS_BUCKET_NAME = f"bucket_{DAG_ID}_{ENV_ID}"
+DATA_SAMPLE_GCS_OBJECT_NAME = "bigquery/us-states/us-states.csv"
+CSV_FILE_LOCAL_PATH = str(Path(__file__).parent / "resources" / "us-states.csv")
+
+
+with models.DAG(
+    DAG_ID,
+    schedule_interval="@once",
+    start_date=datetime(2021, 1, 1),
+    catchup=False,
+    tags=["example", "bigquery"],
+) as dag:
+    create_bucket = GCSCreateBucketOperator(task_id="create_bucket", bucket_name=DATA_SAMPLE_GCS_BUCKET_NAME)
+
+    create_dataset = BigQueryCreateEmptyDatasetOperator(task_id="create_dataset", dataset_id=DATASET_NAME)
+
+    upload_file = LocalFilesystemToGCSOperator(
+        task_id="upload_file_to_bucket",
+        src=CSV_FILE_LOCAL_PATH,
+        dst=DATA_SAMPLE_GCS_OBJECT_NAME,
+        bucket=DATA_SAMPLE_GCS_BUCKET_NAME,
+    )
+
+    # [START howto_operator_bigquery_create_external_table]
+    create_external_table = BigQueryCreateExternalTableOperator(
+        task_id="create_external_table",
+        destination_project_dataset_table=f"{DATASET_NAME}.external_table",
+        bucket=DATA_SAMPLE_GCS_BUCKET_NAME,
+        source_objects=[DATA_SAMPLE_GCS_OBJECT_NAME],
+        schema_fields=[
+            {"name": "emp_name", "type": "STRING", "mode": "REQUIRED"},
+            {"name": "salary", "type": "INTEGER", "mode": "NULLABLE"},
+        ],
+    )
+    # [END howto_operator_bigquery_create_external_table]
+
+    delete_dataset = BigQueryDeleteDatasetOperator(
+        task_id="delete_dataset",
+        dataset_id=DATASET_NAME,
+        delete_contents=True,
+        trigger_rule=TriggerRule.ALL_DONE,
+    )
+
+    delete_bucket = GCSDeleteBucketOperator(
+        task_id="delete_bucket", bucket_name=DATA_SAMPLE_GCS_BUCKET_NAME, trigger_rule=TriggerRule.ALL_DONE
+    )
+
+    (
+        # TEST SETUP
+        [create_bucket, create_dataset]
+        # TEST BODY
+        >> upload_file
+        >> create_external_table
+        # TEST TEARDOWN
+        >> delete_dataset
+        >> delete_bucket
+    )
+
+    list(dag.tasks) >> watcher()
```

Comment:
I know this was discussed int he email thread, but I'll raise the concern one last time because this example really drives it home, then I promise I'll drop it.  We are adding the exact same line of code to every Example DAG (now System Test), shouldn't that really be handled behind the scenes when the test DAG is being parsed instead of adding this copy/pasted line to literally hundreds of files?

Reply:
I don't have a better idea how this can be achieved without explicitly defining this task dependency. This is actually quite clever because in the previous version of this design we were listing all the tasks again and then defining `>>` operator with the watcher. Now it's just this one line that has a reference to all tasks in a local dag and then appends the watcher to all of them. And it is of course only needed when there is a teardown task in the dag.
But of course I am open to any specific suggestions that may improve the design.

Reply:
We thought about a special "util" function for that.

I do not think we discussed about "magical" adding it. this is a bit of a hassle, I agreee but adding it would likely require something on the DAG level - for example a special parameter or decorator. And the problem with that is that it would have to be added somewhere at the beginning of the DAG - where you define the DAG most likely. And the problem with that is that you also use the DAG as examples in our documentation. And we extracts parts of the examples into the documentaiton and we should not pollute those examples with things that are not really good "examples" on how you should add your DAGs. - and those example usually show the DAG definition/default_args as part of the example - by having the special decorator, or parameter on the DAG to indicate that DAG shoudl have "watcher" added migth be too easily copied from those extracted examples.

Having explicit watcher makes it so much easier - it is at the and and it is explicit (which means there is no magic)

But actually what made me think now - we should actually make it even more separated an explicit.

@mnojek why don't we change the the watcher's local import and some comment there to make it even more separated and "explicit" for example:

```
         >> delete_bucket
    ) 
    
    from tests.system.utils.watcher import watcher
    # This test run as a system test needs watcher in order to mark success/failure
    # where "tearDown" task is part of the DAG
    list(dag.tasks) >> watcher()
    
```

Reply:
> I am open to any specific suggestions

I get that.  Unfortunately I only have half-baked suggestions.  the only thing I can really think of would be if we had something like `tests.system.DAG` which inherits `airflow.DAG` and adds that watcher to the task list, but that's only a half-baked idea and I haven't really thought it through all the way.  Writing the same line in hundreds of files just feels wrong, but without a more solid alternative, I'll stop beating that dead horse.  Thanks for the effort you've put into this, and I hope I'm not coming off as argumentative or anything.

Reply:
> Writing the same line in hundreds of files just feels wrong
I fully agree with you and it was actually the reason for the redesign - where we had dozens of lines repeated for every test only to trigger other file with the DAG. And we always welcome any suggestions and discussion. We changed the design several times over the time thanks for the input from the others. 
I agree with Jarek here - the goal is to have an example dag that can be used by other Airflow users as reference and it would be best to avoid any code specific only to system test. Proposed local import of the watcher looks like a good idea for me for this reason alone

Reply:
I support the idea about moving the import for the watcher to local scope. It will be more obvious why it is needed, and we can also add the comment that @potiuk suggested (or similar).
And yeah, copy-pasting the same piece of code in many places is not the best, but at least here this "piece" is small and it's better to have it explicitly than doing some magic stuff that will only shadow the real implementation.
I will do the changes next week.

Reply:
Updated the code and moved the watcher import closer its usage.
Also added the comment to demystify this piece of code.
Please check and confirm if it's fine now 😃

Reply:
We can also configure a cluster policy that will perform the necessary steps only in tests.
Here are docs about cluster policy: 
https://airflow.apache.org/docs/apache-airflow/stable/concepts/cluster-policies.html?highlight=cluster%20policy#dag-policies

Reply:
Cluster policy looks nice but it's overkill in our case when after discussion in this PR we managed to store the watcher together in the test method. It's worth to remember though in case we will need any extra steps.